### PR TITLE
PAPP-20822 Update Crowdstrikeoauth get device scroll widget

### PR DESCRIPTION
**Pull request by gary-phantom**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)
Screen Shot 2022-05-03 at 1 49 35 PM

---
Thanks for contributing!

### CODE REVIEW

File: crowdstrikeoauthapi.json [hunk 1]
```diff
@@ -14808,10 +14811,9 @@
             ],
             "versions": "EQ(*)",
             "render": {
-                "width": 12,
-                "title": "Device scrolls",
-                "type": "table",
-                "height": 5
+                "title": "Device scroll",
+                "type": "custom",
+                "view": "crowdstrike_view.display_view"
```
Comment (paulnawat) [inline crowdstrikeoauthapi.json:14816]: Seeing that the custom view is using `<table>`, would it be simpler to use the default table?

File: crowdstrike_get_device_scroll.html [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        <h4 class="wf-h4-style">No data found</h4>
+
+    {% else %}
+    <h3 class="wf-h3-style">Info</h3>
+    {% if result.check_param %}
+        <table class="wf-table-vertical">
+        {% if result.param.filter %}
+            <tr>
+                <td>Filter</td>
+                <td>{{ result.param.filter }}</td>
```
Comment (paulnawat) [inline crowdstrike_get_device_scroll.html:117]: The displayed values are just `string`, right?

**Review by paulnawat — COMMENTED** [review]
Changes look great. I have a few questions about the custom view.

File: crowdstrikeoauthapi.json (hunk 1 above)
Comment (gary-phantom) [reply crowdstrikeoauthapi.json:14816]: I considered using default table but other actions that also use similar parameters like sort, limit used the custom view as well which show the the parameters as another widget. 
Screen Shot 2022-05-03 at 1 49 35 PM

File: crowdstrike_get_device_scroll.html (hunk 2 above)
Comment (paulnawat) [reply crowdstrike_get_device_scroll.html:117]: Resolved as per our discussion on Slack

File: crowdstrikeoauthapi.json (hunk 1 above)
Comment (paulnawat) [reply crowdstrikeoauthapi.json:14816]: Resolved as per our discussion on Slack

**Comment by gary-phantom** [comment]
Integration test is failing on random playbooks and there are 29 different playbooks involved. 
I re-ran multiple times and they show random failures as well. 
The latest re-run shows a single playbook failure which passes fine when run manually. 
@paulnawat can we force-merge?

test result: https://jenkins.engineering.phantom.us:8443/job/run-app-integration-tests/7610/console
single playbook run: https://10.1.66.16/mission/1315/analyst/timeline



Screen Shot 2022-05-04 at 1 31 07 PM

**Comment by paulnawat** [comment]
@gary-phantom From the attached error message, the integration tests are failing due to the 1hr Github limit.
In this case, we would use a scheduled pipeline where there is no time limit. Could you give that a try?

You can then attach the successful scheduled pipeline in this PR. This is how we ensure pipeline is green before it gets merged to `next`

see [reference](https://github.com/splunk-soar-connectors/windowsdefenderatp/pull/8)

File: crowdstrikeoauthapi.json [hunk 3]
```diff
@@ -15,11 +15,11 @@
         }
     ],
     "license": "Copyright (c) 2019-2022 Splunk Inc.",
-    "app_version": "3.6.1",
+    "app_version": "3.6.0",
```
Comment (paulnawat) [inline crowdstrikeoauthapi.json:18]: The version should be bumped up instead of decrementing, right?

Comment (gary-phantom) [reply crowdstrikeoauthapi.json:18]: the version got incremented accidentally. so I reverted it back to original 3.6.0

Comment (paulnawat) [reply crowdstrikeoauthapi.json:18]: I see. Thanks!

**Comment by gary-phantom** [comment]
@paulnawat scheduled run on gary/PAPP-20822-crowdstrikeoauth passed 
https://cd.splunkdev.com/phantom-apps/app-tests/-/pipelines/5354063
Screen Shot 2022-05-04 at 2 41 35 PM

**Review by paulnawat — APPROVED** [review]

**Review by mpan-splunk — APPROVED** [review]

**Pull request merged**